pkg/pool-manager/pool_test.go

```diff
@@ -589,4 +590,182 @@ var _ = Describe("Pool", func() {
+					Name: webhookName,
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							metav1.LabelSelectorRequirement{
+								Key:      "runlevel",
+								Operator: "NotIn",
+								Values:   []string{"0", "1"},
+							},
+							metav1.LabelSelectorRequirement{
+								Key:      "openshift.io/run-level",
+								Operator: "NotIn",
+								Values:   []string{"0", "1"},
+							},
+							metav1.LabelSelectorRequirement{
+								Key:      LabelKey,
+								Operator: "NotIn",
+								Values:   []string{excludingValue},
+							},
+						},
+					},
+				},
+			},
+		}
+		optInMutatingWebhookConfiguration := &v1beta1.MutatingWebhookConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: optInMutatingWebhookConfigurationName,
+			},
+			Webhooks: []v1beta1.MutatingWebhook{
+				v1beta1.MutatingWebhook{
+					Name: webhookName,
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							metav1.LabelSelectorRequirement{
+								Key:      "runlevel",
+								Operator: "NotIn",
+								Values:   []string{"0", "1"},
+							},
+							metav1.LabelSelectorRequirement{
+								Key:      "openshift.io/run-level",
+								Operator: "NotIn",
+								Values:   []string{"0", "1"},
+							},
+							metav1.LabelSelectorRequirement{
+								Key:      LabelKey,
+								Operator: "In",
+								Values:   []string{includingValue},
+							},
+						},
+					},
+				},
+			},
+		}
+		namespaceWithIncludingLabel := &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   namespaceWithIncludingLabelName,
+				Labels: map[string]string{LabelKey: includingValue},
+			},
+		}
+		namespaceWithExcludingLabel := &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   namespaceWithExcludingLabelName,
+				Labels: map[string]string{LabelKey: excludingValue},
+			},
+		}
+		namespaceWithNoLabels := &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceWithNoLabelsName,
+			},
+		}
+		namespaceWithIrrelevantLabels := &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   namespaceWithIrrelevantLabelsName,
+				Labels: map[string]string{"other": "label"},
+			},
+		}
+		var poolManager *PoolManager
+		BeforeEach(func() {
+			poolManager = createPoolManager("02:00:00:00:00:00", "02:00:00:00:00:01", optOutMutatingWebhookConfiguration, optInMutatingWebhookConfiguration, namespaceWithIncludingLabel, namespaceWithExcludingLabel, namespaceWithNoLabels, namespaceWithIrrelevantLabels)
+		})
+		table.DescribeTable("Should return the expected namespace acceptance outcome according to the opt-mode",
+			func(n *compareNamespaceSelectorToOptModeParams) {
+				isNamespaceManaged, err := poolManager.compareNamespaceSelectorToOptMode(n.namespaceName, n.mutatingWebhookConfigurationName, webhookName, n.optMode)
+				Expect(err).ToNot(HaveOccurred(), "compareNamespaceSelectorToOptMode should not return error")
```

Comment:
There are multiple exit points with error. I think it is not checked anywhere.

Reply:
you are right. 
Added the error scenarios.
Done